File: lib/java/openlinktoken-core-ai/src/main/java/org/openlinktoken/tokens/definitions/ML1Token.java
/* SPDX-License-Identifier: MIT */
package org.openlinktoken.tokens.definitions;

import java.util.ArrayList;

import org.openlinktoken.attributes.AttributeExpression;
import org.openlinktoken.tokens.Token;

/**
 * Represents optional ML1 token definition.
 */
public class ML1Token implements Token {
    private static final long serialVersionUID = 1L;
    private static final String ID = "ML1";

    private final ArrayList<AttributeExpression> definition = new ArrayList<>();

    @Override
    public String getIdentifier() {
        return ID;
    }

    @Override
    public ArrayList<AttributeExpression> getDefinition() {
        return definition;
    }
}

File: lib/python/openlinktoken-core-ai/src/main/openlinktoken_core_ai/tokens/ml1_token.py
"""Optional ML1 token definition."""
from typing import List

from openlinktoken.attributes.attribute_expression import AttributeExpression
from openlinktoken.tokens.token import Token


class ML1Token(Token):
    """Represents optional ML1 token definition."""

    ID = "ML1"

    def __init__(self):
        self._definition: List[AttributeExpression] = []

    def get_identifier(self) -> str:
        return self.ID

    def get_definition(self) -> List[AttributeExpression]:
        return self._definition
